### PR TITLE
Update look and feel copy

### PR DESCRIPTION
**Pull request by WiNloSt**
Part of #44273

I initially intended for this PR to implement the toggle for #44273, but after the initial copy change, it seemed it involved a lot of line changes, so I'll just leave it here.

This PR updates the copy on the "Look and Feel" tab (was "Appearance") and reorders the location of each setting + renames some labels to [match the design](https://www.figma.com/design/1YcsFf5yYdFZTguwoKVdkj/Re-design-the-transparent-background-setting-in-interactive-embedds?node-id=917-6518&t=NKXLRBgBaxzBbFQe-0). 

No new tests has been added, since this is only changing copies, only failed existing tests are fixed.

### CODE REVIEW

File: e2e/test/scenarios/sharing/sharing-reproductions.cy.spec.js [hunk 1]
```diff
@@ -630,15 +630,15 @@ describeEE("issue 26988", () => {
...
+      activeTab: "lookAndFeel",
       previewMode: "preview",
       acceptTerms: false,
     });
 
     cy.wait("@previewDashboard");
     getIframeBody().should("have.css", "font-family", "Lato, sans-serif");
 
-    cy.findByLabelText("Playing with appearance options")
+    cy.findByTestId("look-and-feel-settings")
```
Comment (WiNloSt) [inline e2e/test/scenarios/sharing/sharing-reproductions.cy.spec.js:641]: The reason this is changed is because the new design removes the header entirely (there used to be a header in the red circle), so I need to get it by something else. I think `testid` is appropriate in this situation.
Screenshot 2024-06-17 at 9 41 07 PM

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx [hunk 2]
```diff
@@ -1,9 +1,8 @@
-import type { ChangeEvent } from "react";
+import { match } from "ts-pattern";
```
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx:1]: All changes are to match the new design minus the move of transparent theme to its toggle, that will be another PR.

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx [hunk 3]
```diff
@@ -214,3 +221,11 @@ function getTitleLabel(resourceType: EmbedResourceType) {
 
   return null;
 }
+
+function getBorderTitle(resourceType: EmbedResourceType) {
+  return match(resourceType)
```
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx:226]: Use the new snazzy `ts-pattern`

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/StaticEmbedSetupPane.tsx [hunk 4]
```diff
@@ -226,7 +226,7 @@ export const StaticEmbedSetupPane = ({
     const locationMap = {
       overview: "code_overview",
       parameters: "code_params",
-      appearance: "code_appearance",
+      lookAndFeel: "code_appearance",
```
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/StaticEmbedSetupPane.tsx:229]: `"code_appearance"` value is kept as is since it's used in analytics.

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/common.unit.spec.tsx [hunk 5]
```diff
@@ -477,7 +475,7 @@ describe("Static Embed Setup phase", () => {
         ).toBeVisible();
 
         const link = within(
-          screen.getByLabelText("Playing with appearance options"),
+          screen.getByTestId("look-and-feel-settings"),
```
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/common.unit.spec.tsx:478]: There used to be a header in the red circle, but we remove it in the new design
Screenshot 2024-06-17 at 9 41 07 PM

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/premium.unit.spec.tsx [hunk 6]
```diff
@@ -54,7 +54,7 @@ describe("Static Embed Setup phase - EE, with token", () => {
 
         await userEvent.click(fontSelect);
 
-        const popover = await screen.findByRole("grid");
+        const popover = await screen.findByRole("listbox", { name: "Font" });
```
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/premium.unit.spec.tsx:57]: We used the old `Select` component which renders different content, now we use Mantine's `Select` so I need to change the query to match the new select popover.

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx (hunk 2 above)
Comment (heypoom) [inline frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx:1]: nice use of `ts-pattern` here! 👍🏻

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx [hunk 7]
```diff
@@ -59,26 +65,61 @@ export const AppearanceSettings = ({
...
+        <Text>{jt`These options require changing the server code. You can play around with and preview the options here. Check out the ${(
           <ExternalLink
             key="doc"
             href={`${docsUrl}${utmTags}#customizing-the-appearance-of-static-embeds`}
           >{t`documentation`}</ExternalLink>
         )} for more.`}</Text>
       </StaticEmbedSetupPaneSettingsContentSection>
       <StaticEmbedSetupPaneSettingsContentSection
-        title={t`Playing with appearance options`}
         mt="2rem"
```
Comment (heypoom) [inline frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx:83]: design-wise, do we still need the margin-top here if there is no title? (this is out-of-scope for this PR, but just a reminder)

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx [hunk 8]
```diff
@@ -59,26 +65,61 @@ export const AppearanceSettings = ({
...
+                </Text>
+              }
+              value={
+                displayOptions.font === null ? "null" : displayOptions.font
+              }
+              data={[
+                {
+                  label: t`Use instance font`,
+                  value: "null",
+                },
```
Comment (heypoom) [inline frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx:101]: is using the `"null"` string as a default select value for "use instance font" here intended? i can see that this is used as a fallback option in `onChangeDisplayOptions` below, but that jumped out at me a bit as I thought it's a mistake.

no need to change this though.

**Review by heypoom — APPROVED** [review]
Looks great!

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx (hunk 8 above)
Comment (WiNloSt) [reply frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx:101]: I forgot to add a comment here. I wish I could use `null`. Technically that works, but it will fail TypeScript check since somehow Mantine decided the value should only be a string. Maybe using `null` with ts ignore comment is a better solution here, any thoughts?

https://github.com/mantinedev/mantine/blob/311a893107719d450a88f4a0df3c161e9911a5e0/src/mantine-core/src/Select/types.ts#L11

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx [hunk 9]
```diff
@@ -59,26 +65,60 @@ export const AppearanceSettings = ({
...
+                <Text fw="bold" mb="0.25rem" lh="1rem">
+                  Font
+                </Text>
+              }
+              value={displayOptions.font}
+              data={[
+                {
+                  label: t`Use instance font`,
+                  // @ts-expect-error Mantine v6 and v7 both expect value to be a string
+                  value: null,
```
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx:95]: @heypoom I reverted the value back to `null` as before, but added a comment explaining why I need to disable TypeScript check here instead. I think it's cleaner than trying to force it to be string as Mantine expects.

File: e2e/test/scenarios/sharing/sharing-reproductions.cy.spec.js [hunk 10]
```diff
@@ -630,15 +630,15 @@ describeEE("issue 26988", () => {
...
+      activeTab: "lookAndFeel",
       previewMode: "preview",
       acceptTerms: false,
     });
 
     cy.wait("@previewDashboard");
     getIframeBody().should("have.css", "font-family", "Lato, sans-serif");
 
-    cy.findByLabelText("Playing with appearance options")
+    cy.findByLabelText("Customizing look and feel")
```
Comment (WiNloSt) [inline e2e/test/scenarios/sharing/sharing-reproductions.cy.spec.js:641]: Since the removal of the second section, all the settings belong to the first section now.

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/common.unit.spec.tsx [hunk 11]
```diff
@@ -477,7 +475,7 @@ describe("Static Embed Setup phase", () => {
         ).toBeVisible();
 
         const link = within(
-          screen.getByLabelText("Playing with appearance options"),
+          screen.getByLabelText("Customizing look and feel"),
```
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/common.unit.spec.tsx:478]: Since the removal of the second section, all the settings belong to the first section now.

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx [hunk 12]
```diff
@@ -59,26 +65,56 @@ export const AppearanceSettings = ({
...
-          >{t`documentation`}</ExternalLink>
-        )} for more.`}</Text>
-      </StaticEmbedSetupPaneSettingsContentSection>
-      <StaticEmbedSetupPaneSettingsContentSection
-        title={t`Playing with appearance options`}
-        mt="2rem"
+        title={t`Customizing look and feel`}
       >
         <Stack spacing="1rem">
+          <Text>{jt`These options require changing the server code. You can play around with and preview the options here. Check out the ${(
```
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx:76]: Slightly modify the markup to match the parameter tab.

https://github.com/metabase/metabase/blob/bdae7827e4cba3114420f41307160d324eec5c5b/frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/ParametersSettings.tsx#L63-L66

So in the section, we start off with
1. Stack (This separates the description and settings with a space)
2. Text (description)
3. A list of settings

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx (hunk 8 above)
Comment (WiNloSt) [reply frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx]: Fixed in https://github.com/metabase/metabase/pull/44277/commits/d2652b026b7cecd831bff6bc136109668c96dfe2
